pkg/backup/backup_compaction.go

```diff
@@ -653,3 +703,71 @@ func processProgress(
 	}
 	return nil
 }
+
+type compactionChain struct {
+	// restoreChain is the linear chain of backups up to the end time required
+	// for a restore.
+	restoreChain   []backuppb.BackupManifest
```

Comment:
nit: i thinks this should still be called backupChain, as this represents a chain of backups. if we ever (god forbid) need to load up all backups in a collection, we'll use the `backuptree` var name.